pkg/agent/cniserver/server.go

```diff
@@ -558,12 +569,34 @@ func (s *CNIServer) Initialize(
 	entityUpdates chan<- types.EntityReference,
 ) error {
 	var err error
-	s.podConfigurator, err = newPodConfigurator(
+	s.PodConfigurator, err = newPodConfigurator(
+		ovsBridgeClient, ofClient, s.routeClient, ifaceStore, s.nodeConfig.GatewayConfig.MAC,
+		ovsBridgeClient.GetOVSDatapathType(), ovsBridgeClient.IsHardwareOffloadEnabled(), entityUpdates,
+	)
+	if err != nil {
+		return fmt.Errorf("error during initialize PodConfigurator: %v", err)
+	}
+	if err := s.reconcile(); err != nil {
+		return fmt.Errorf("error during initial reconciliation for CNI server: %v", err)
+	}
+	return nil
+}
+
+func (s *CNIServer) InitializeWithCniCache(
+	ovsBridgeClient ovsconfig.OVSBridgeClient,
+	ofClient openflow.Client,
+	ifaceStore interfacestore.InterfaceStore,
+	entityUpdates chan<- types.EntityReference,
+	cniPodInfoStore cnipodcache.CniPodInfoStore,
+) error {
+	var err error
+	s.PodConfigurator, err = newPodConfiguratorWithCniCache(
```

Comment:
Again, could we just add cniPodInfoStore argument to newPodConfigurator() instead of adding new func with code duplication?